src/cargo/core/compiler/context/compilation_files.rs

```diff
@@ -622,7 +628,8 @@ fn should_use_metadata(bcx: &BuildContext<'_, '_>, unit: &Unit) -> bool {
     if (unit.target.is_dylib()
         || unit.target.is_cdylib()
         || (unit.target.is_executable() && short_name.starts_with("wasm32-"))
-        || (unit.target.is_executable() && short_name.contains("msvc")))
+        || (unit.target.is_executable() && short_name.contains("msvc"))
+        || (unit.target.is_executable() && short_name.contains("-apple-")))
```

Comment:
Should this maybe be changed to a `CfgExpr` test?  I'm becoming more uneasy with anything that inspects the target triple, since that doesn't work for JSON targets, and generally seems a little sloppy.